pkg/sentry/socket/hostinet/socket.go

```diff
@@ -406,10 +406,16 @@ func (s *socketOpsCommon) GetSockOpt(t *kernel.Task, level int, name int, optVal
 		}
 	case linux.SOL_TCP:
 		switch name {
-		case linux.TCP_NODELAY:
+		case linux.TCP_NODELAY, linux.TCP_MAXSEG:
 			optlen = sizeofInt32
 		case linux.TCP_INFO:
 			optlen = linux.SizeOfTCPInfo
+			// tcp_info length should equal to user required due to user process may be compiled with lower kernel version
+			if optlen > outLen {
+				optlen = outLen
+			}
```

Comment:
This change seems unrelated to this PR and also not sure if we want to do this. If we want to do this, we should do it separately.

On older kernel versions, what outLen will be passed by the user?

Reply:
This is also an error of iperf3 from image 'networkstatic/iperf3', if we use strace to trace syscall getsockopt when running 'iperf3 -c', we will get EINVAL like this:
getsockopt(5, SOL_TCP, TCP_INFO, 0x7f7b4d716a14, [104]) = -1 EINVAL (Invalid argument)
iperf3 passed outLen 104, and linux.SizeOfTCPInfo is 224 currently, so gvisor returned EINVAL.
Actually the linux kernel and sandboxnet of gvisor have the same logic, see 
https://github.com/google/gvisor/blob/master/pkg/sentry/socket/netstack/netstack.go#L1212, buf is truncated by outLen

Reply:
I see, thanks for the code pointer.

I still think this should be done as a separate PR, because this PR is about adding support for TCP_MAXSEG, TCP_CONGESTION, not fixing TCP_INFO.

Reply:
OK, I'll fix this in new PR